### PR TITLE
Sublists: generalize disjoint-union-is-cospan to upper-bound-is-cospan

### PR DESCRIPTION
**Pull request by andreasabel**
The disjointness assumption is superfluous.

### CODE REVIEW

File: src/Data/List/Relation/Binary/Sublist/Propositional.agda [hunk 1]
```diff
@@ -152,3 +152,68 @@ separateˡ (z  ∷ʳ τ₁) (z  ∷ʳ τ₂) = z   ∷ₙ-Sep separateˡ τ₁ 
 separateˡ (z  ∷ʳ τ₁) (y≡z ∷ τ₂) = y≡z ∷ᵣ-Sep separateˡ τ₁ τ₂
 separateˡ (x≡z ∷ τ₁) (z  ∷ʳ τ₂) = x≡z ∷ₗ-Sep separateˡ τ₁ τ₂
 separateˡ (x≡z ∷ τ₁) (y≡z ∷ τ₂) = ∷-Sepˡ x≡z y≡z (separateˡ τ₁ τ₂)
+
+------------------------------------------------------------------------
+-- A Union where the triangles commute is a
+-- Cospan in the slice category (_ ⊆ zs).
```
Comment (andreasabel) [inline src/Data/List/Relation/Binary/Sublist/Propositional.agda:158]: Should this moved content rather go into a new module `...Sublist.Propositional.Slice`?

Comment (MatthewDaggitt) [reply src/Data/List/Relation/Binary/Sublist/Propositional.agda:158]: Yup that seems sensible!

File: src/Data/List/Relation/Binary/Sublist/Propositional/Disjoint.agda [hunk 2]
```diff
@@ -1,34 +1,31 @@
 ------------------------------------------------------------------------
 -- The Agda standard library
 --
--- Sublist-related properties
+-- This module is DEPRECATED.
+-- Please use `Data.List.Relation.Binary.Sublist.Propositional` instead.
```
Comment (MatthewDaggitt) [inline src/Data/List/Relation/Binary/Sublist/Propositional/Disjoint.agda:5]: This comment will need to change as well?

**Comment by andreasabel** [comment]
I implemented the discussed changes.

**Review by MatthewDaggitt — COMMENTED** [review]
Okay great, thanks! All that's needed is a CHANGELOG entry and I'll merge it in.

**Comment by andreasabel** [comment]
I added the changelog entries.

I noticed that CHANGELOG.md has whitespace violations.  Why is it except from `fix-whitespace`?

**Comment by MatthewDaggitt** [comment]
> I noticed that CHANGELOG.md has whitespace violations. Why is it except from fix-whitespace?

Don't know. Just opened a PR to fix it.

**Review by MatthewDaggitt — APPROVED** [review]

File: standard-library.agda-lib [hunk 3]
```diff
@@ -1,4 +1,5 @@
-name: standard-library-2.1
+name: standard-library-tbt
 include: src
 flags:
   --warning=noUnsupportedIndexedMatch
+--   --type-based-termination
```
Comment (jamesmckinna) [inline standard-library.agda-lib:5]: Was this meant to be part of this PR? 
AFAICT, this is part of a separate experiment on the main `Agda` line of development, and not relevant (yet!) for `stdlib-v2.1`?
Should the commit which introduced this be reverted?

Comment (andreasabel) [reply standard-library.agda-lib:5]: Aarrgh how did this happen?  Sincere apologies, will fix.

Comment (andreasabel) [reply standard-library.agda-lib:5]: https://github.com/agda/agda-stdlib/pull/2326

**Pull request merged**